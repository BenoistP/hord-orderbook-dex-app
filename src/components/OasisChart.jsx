import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import tokens from './../store/selectors/tokens';

import OasisWidgetFrame from '../containers/OasisWidgetFrame';
import OasisChartPrice from './OasisChartPrice';
import OasisChartVolume from './OasisChartVolume';
import OasisChartDepth from './OasisChartDepth';

// import styles from './OasisChart.scss';

const propTypes = PropTypes && {};
const defaultProps = {};

export class OasisChart extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {chart: 'PRICE'};
    this.chartChanged = this.chartChanged.bind(this);
  }

  chartChanged(e) {
    this.setState({chart: e.target.value});
  }

  render() {
    const types = ['PRICE', 'VOLUME', 'DEPTH'];
    return (
        <OasisWidgetFrame heading="Charts">
          <select value={this.state.chart} onChange={this.chartChanged}>
            { types.map(t => <option key={t}>{t}</option>) }
          </select>
          <div>
            { this.state.chart == "PRICE" && <OasisChartPrice tradingPair={this.props.activeTradingPair}/> }
            { this.state.chart == "VOLUME" && <OasisChartVolume tradingPair={this.props.activeTradingPair}/> }
            { this.state.chart == "DEPTH" && <OasisChartDepth tradingPair={this.props.activeTradingPair}/> }
          </div>
        </OasisWidgetFrame>
    );
  }
}

export function mapStateToProps(state) {
  return {
    activeTradingPair: tokens.activeTradingPair(state),
  };
}

export function mapDispatchToProps(dispatch) {
  const actions = {
  };
  return { actions: bindActionCreators(actions, dispatch) };
}

OasisChart.displayName = 'OasisChart';
OasisChart.propTypes = propTypes;
OasisChart.defaultProps = defaultProps;
export default connect(mapStateToProps, mapDispatchToProps)(OasisChart);
