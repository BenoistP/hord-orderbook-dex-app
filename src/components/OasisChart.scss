.select {
  float: right;
  position: relative;
  top: -1px;
}
