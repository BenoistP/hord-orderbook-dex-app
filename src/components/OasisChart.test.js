/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import OasisChart from './OasisChart';


describe('(Component) OasisChart', () => {
  it('should render', () => {
    const props = {};
    const wrapper = shallow(
      <OasisChart {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
