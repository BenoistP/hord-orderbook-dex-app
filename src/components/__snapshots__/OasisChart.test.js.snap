// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisChart should render 1`] = `
<OasisWidgetFrame
  className=""
  heading="Charts"
  headingChildren={
    <OasisSelect
      className={undefined}
      onChange={[Function]}
      size="md"
      value="PRICE"
    >
      <option>
        PRICE
      </option>
      <option>
        VOLUME
      </option>
      <option>
        DEPTH
      </option>
    </OasisSelect>
  }
>
  <div>
    <Connect(OasisChartPrice)
      tradingPair={
        Object {
          "baseToken": "MKR",
          "quoteToken": "DAI",
        }
      }
    />
  </div>
</OasisWidgetFrame>
`;
